mycore-mods/src/main/java/org/mycore/mods/merger/MCRAbstractRedundantModsEventHandler.java

```diff
@@ -0,0 +1,172 @@
+package org.mycore.mods.merger;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jdom2.Element;
+import org.mycore.common.MCRConstants;
+import org.mycore.common.events.MCREvent;
+import org.mycore.common.events.MCREventHandlerBase;
+import org.mycore.datamodel.classifications2.MCRCategoryID;
+import org.mycore.datamodel.metadata.MCRObject;
+import org.mycore.mods.MCRMODSWrapper;
+import org.mycore.mods.classification.MCRClassMapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Checks for and removes redundant classifications in Mods-Documents. If a classification category and
+ * the classification's child category are both present in the document, the parent classification will
+ * be removed. <br><br>
+ * This abstract class can be extended by overriding the following abstract methods:
+ * <ol>
+ *     <li>{@link MCRAbstractRedundantModsEventHandler#isConsistent(Element, Element)}: custom checks for
+ *     the two classifications' consistency.</li>
+ *     <li>{@link MCRAbstractRedundantModsEventHandler#getClassificationElementName()}: the name of
+ *     the mods-element that the EventHandler is checking duplicates for.</li>
+ * </ol>
+ */
+public abstract class MCRAbstractRedundantModsEventHandler extends MCREventHandlerBase {
+
+    private final Logger logger = LogManager.getLogger(getClass());
```

Comment:
remove `getClass()`

Reply:
This is an abstract class, afaik the `getClass()` is necessary to initialize the logger with the actual class (that extends from this abstract class). This is also the reason, why this logger is not static. Without this, interpreting logging messages can be relay challenging, because from the log messages, it is not discernable which handler is actually executing.

Reply:
my fault..